mmv1/third_party/terraform/utils/framework_provider.go.erb

```diff
@@ -37,17 +38,19 @@ func New(version string) provider.ProviderWithMetaSchema {
 }
 
 // frameworkProvider is the provider implementation.
-type frameworkProvider struct{
+type frameworkProvider struct {
     billingProject types.String
     client *http.Client
     context context.Context
     gRPCLoggingOptions []option.ClientOption
     pollInterval time.Duration
     project types.String
-    region string
+    region types.String
+    zone types.String
     requestBatcherIam *RequestBatcher
     requestBatcherServiceUsage *RequestBatcher
     scopes []string
+    tokenSource oauth2.TokenSource
```

Comment:
This follows the example of : https://github.com/hashicorp/magic-modules/blob/19b676e971d55e8524600bb33ae62a10d43b223c/mmv1/third_party/terraform/utils/config.go.erb#L194